Range_Checker.c

```diff
@@ -0,0 +1,79 @@
+#include"stdio.h"
+#include "Range_checker.h"
+
+void rangeCheck(float *dataset_arr, unsigned int length_dataset)
+{
+	static int idx = 0, range_start_idx = 0, range_end_idx = 0; 
+	float diff ,range_start, max_same_range_diff = 2.0;
+	int pattern_found = 0;
+	int pattern_started = 0;
+	if(sanityCheck( dataset_arr, length_dataset ) == 1)
+	{
+		for(idx = 1 ; idx <length_dataset ; idx ++ )
+		{
+			diff = DIFFERENCE_CHECK(dataset_arr[idx-1] , dataset_arr[idx]);
+			if(diff < max_same_range_diff)
+			{				
+				/**update the new range start index**/
+				range_end_idx = idx ; // Update the index
+				pattern_started = 1;
+				pattern_found = 1;
+			}
+			else
+			{
+				pattern_started = 0;
+			}
+			if( idx == (length_dataset-1))//last loop. So making the previous pattern end.
+			{
+				pattern_started = 0;
+			}
+			if(pattern_started == 0) /*only print if pattern sequence is not present*/
+			{
+				(void)Print_Range(range_start_idx , range_end_idx , dataset_arr , pattern_found, UPDATE);
+				range_start_idx = idx ; // Update the index
+			}
+		}
+		/**Be ready for next call- This Error found after failing test case**/
+		idx = 0;
+		range_start_idx = 0;
+		range_end_idx = 0;
+		/*************************************/
+	}
+}
+
+int Print_Range(int start_idx , int end_idx , float* data_buff, int range_found, int objective)
+{
+	static int print_done = 0;

```

Comment:
Since `print_done` is static, it will remember its previous value. When `objective` is not UPDATE, the retVal will take the previous value of `print_done`. This makes the pass/fail depend on the order of test execution, which can be different across runs.
Recommendations:
1. Don't use static
2. Use the name of the variable to convey what it signifies. Dont use `retVal`